### ISSUE TITLE
subtract_self_energies returns empty TransformableIterable

### ISSUE THREAD

**Issue by ajsmit24** [aiqm/torchani#517]
I am attempting to train the ANI model based on some in house data and following [this](https://github.com/aiqm/torchani/blob/master/examples/nnp_training.py) tutorial:

My problem is at this point:
training, validation = torchani.data.load(dspath).subtract_self_energies(energy_shifter,species_order).species_to_indices(species_order).shuffle().split(0.8, None)

It works fine if I remove subtract_self_energies. What I have determined is that subtract_self_energies is returning an empty TransformableIterable. Notably, my data is coming in the form of jsons that have been turned into a generator of dicts which is turned into a TransformableIterable. The dictionaries are of the form below:

```
{'coordinates': array([[-2.879155e+00, -2.659750e-01,  6.010000e-04],
        [-2.694267e+00,  9.670450e-01, -2.199000e-03],
        [-1.425789e+00,  1.282788e+00, -2.415000e-03],
        [-6.712590e-01,  2.823620e-01, -1.133000e-03],
        [-1.519620e+00, -1.044127e+00,  2.595000e-03],
        [ 6.712590e-01,  2.823620e-01,  1.143000e-03],
        [ 1.425789e+00,  1.282788e+00,  2.416000e-03],
        [ 2.694267e+00,  9.670450e-01,  2.195000e-03],
        [ 2.879155e+00, -2.659750e-01, -6.080000e-04],
        [ 1.519620e+00, -1.044127e+00, -2.594000e-03],
        [-3.750497e+00, -7.489210e-01,  1.302000e-03],
        [-3.435114e+00,  1.635729e+00, -3.818000e-03],
        [-1.089719e+00,  2.221911e+00, -4.189000e-03],
        [ 1.089719e+00,  2.221911e+00,  4.191000e-03],
        [ 3.435114e+00,  1.635729e+00,  3.810000e-03],
        [ 3.750497e+00, -7.489210e-01, -1.316000e-03]]),
 'species': ['C',
  'C',
  'C',
  'C',
  'S',
  'C',
  'C',
  'C',
  'C',
  'S',
  'H',
  'H',
  'H',
  'H',
  'H',
  'H'],
 'energies': -1154.52692963}
```

The dicts are converted from an array to a generator by the following code:
```
ANIs=[]
    for molN in molNames:
        temp=#a method that reads a json to a dict and parses it to the above form
        ANIs.append(temp)
    g=(o for o in ANIs)
    return TransformableIterable(IterableAdapter(lambda: g))
```

Any help would be much appreciated.

**Comment by yueyericardo** [comment]
Hi, the best way to read dataset for torchani is creating a h5 dataset file.
I wrote an example before on how to create dataset, and some basic training codes.
Check at https://github.com/yueyericardo/torchani_training#create-dataset
